include/gfx/timsort.hpp

```diff
@@ -633,6 +634,25 @@ template <typename RandomAccessIterator, typename Compare> class TimSort {
 
 public:
 
+    static void merge(iter_t const lo, iter_t const mid, iter_t const hi, Compare compare) {
+        GFX_TIMSORT_ASSERT(lo <= mid);
+        GFX_TIMSORT_ASSERT(mid <= hi);
+
+        if (lo == mid || mid == hi) {
+            return; // nothing to do
+        }
+
+        TimSort ts;
+        ts.pushRun(lo, mid - lo);
+        ts.pushRun(mid, hi - mid);
+        GFX_TIMSORT_ASSERT(ts.pending_.size() == 2);
+        ts.mergeAt(0, compare);
```

Comment:
This implementation is unoptimized: there is redundant pushing into a vector and redundant checks inside `mergeAt()`. However, it is perfectly good for my use case where `timmerge` is called rarely, but the ranges can be huge. If someone needs to squeeze more performance out of `timmerge`, (s)he can extract the second half of `mergeAt()` into a separate function and call it here. I can do this myself if the reviewer(s) like the idea. But I would rather do this in the next pull request to avoid increasing the diff and prolonging the code review.

Reply:
Turned out that optimizing this actually reduces the diff and noticeably improves performance for `int`s and non-overlapping ranges (*SORTED SEQUENCE*). So I just pushed the optimization and the updated benchmark results into this pull request.